src/native/_auth.js

```diff
@@ -132,12 +132,115 @@ const AuthGranted = Struct({
   bootstrap_config_cap: t.usize
 });
 
-const makeAppInfo = (appInfo) => {
+const toBuffer = (ptr, len) => {
+  return new Buffer(ref.reinterpret(ptr, len, 0))
+}
+
+const makeAppKeys = (appKeys) => {
+  return new AppKeys({
+    owner_key: new t.SIGN_PUBLICKEYBYTES(appKeys.owner_key),
+    enc_key: new t.SYM_KEYBYTES(appKeys.enc_key),
+    sign_pk: new t.SIGN_PUBLICKEYBYTES(appKeys.sign_pk),
+    sign_sk: new t.SIGN_SECRETKEYBYTES(appKeys.sign_sk),
+    enc_pk: new t.ASYM_PUBLICKEYBYTES(appKeys.enc_pk),
+    enc_sk: new t.ASYM_SECRETKEYBYTES(appKeys.enc_sk),
+  });
+}
+
+const makeAccessContInfo = (accessContainer) => {
+  return new AccessContInfo({
+    id: new t.XOR_NAME(accessContainer.id),
+    tag: accessContainer.tag,
+    nonce: new t.SYM_NONCEBYTES(accessContainer.nonce),
+  });
+}
+
+const makeAccessContainerEntry = (accessContainerEntry) => {
+  const contInfoArray = new ContainerInfoArray(accessContainerEntry.length);
+  accessContainerEntry.forEach((entry, index) => {
+    const permissions = new t.PermissionSet({
+      Read: entry.permissions.Read,
+      Insert: entry.permissions.Insert,
+      Update: entry.permissions.Update,
+      Delete: entry.permissions.Delete,
+      ManagePermissions: entry.permissions.ManagePermissions
+    });
+    contInfoArray[index] = new ContainerInfo({
+      name: ref.allocCString(entry.name),
+      mdata_info: helpersForNative.makeMDataInfo(entry.mdata_info),
+      permissions
+    });
+  });
+
+  return new AccessContainerEntry({
+    containers: contInfoArray.buffer,
+    containers_len: contInfoArray.length
+  });
+}
+
+const makeAuthGranted = (authGrantedObj) => {
```

Comment:
is this 'Granting Authentication?' 

To me `AuthGranted` sounds like a boolean as opposed to a class. (not sure what is correct here, it's not that clear that this is doing for me)

Reply:
`AuthGanted` is the name of the Node-FFI struct as defined in safe_app lib, this function just constructs the corresponding FFI-struct from a js object. I'm renaming it to `makeAuthGrantedFfiStruct`.
We can consider some refactoring to make a better naming and differentiation of the functions that convert FFI structs to js objects and vice versa.